Changing how session restart works

houshuang opened the pull request:
This makes two small but important changes to how session restart works. First, it restarts the session from the session graph, not the underlying graph. This means that if you go to the graph editor and makes some small changes, and then restart, these changes will survive. You can always start a new session if you want to "reset" from the initial graph. I think this makes more sense with how we actually work.

Also, before restarting, it checks whether the graph is broken, and does not let you restart if it is. 

Closes #839

houshuang committed:
WIP

lfaucon reviewed (changes requested):
![screenshot from 2018-04-05 11-11-21](https://user-images.githubusercontent.com/8818081/38357219-38df1926-38c2-11e8-812a-e2405f2cb30f.png)

The naming has become crazy, can you update it accordingly to your changes?

houshuang committed:
WIP

houshuang commented:
Good point. I fixed the naming structure, now sessions always have a # first, and then they have a number at the end. I also fixed a bug where the "computed" property wasn't reset on activities.

lfaucon commented on frog/imports/api/sessions.js:
```diff
@@ -185,13 +185,18 @@ const addSessionFn = (graphId: string, slug: string): string => {
-    }).count();
-    const sessionName = '#' + graph.name + ' ' + (count + 1);
-
+    const match = graph.name.match(/(.+)\(([^(]+)\)$/);
```
Should be `/(.+)\((\d+)\)$/`

If I name a graph "hello (world)" the session associated becomes "#hello (NaN)"

houshuang committed:
Merge branch 'develop' into changingRestart

houshuang committed:
WIP

houshuang commented:
@lfaucon Thanks, good idea. Fixed.
